### ISSUE TITLE
Once rendered, Toast color does not change with ColorMode

### ISSUE THREAD

**Issue by remjx** [chakra-ui/chakra-ui#3076]
# 🐛 Bug report

Once Toast is displayed, and user changes ColorMode while Toast is still displayed, the color of the Toast does not change.

## 💥 Steps to reproduce

1. Trigger toast
2. Change color mode
3. Check color of toast

## 💻 Link to reproduction

https://codesandbox.io/s/chakra-ui-toast-colormode-issue-cwroy

## 🧐 Expected behavior

Toast color changes when ColorMode changes

**Comment by dominictwlee** [comment]
Have you tried calling `toast.update` for the particular toast on colorMode change? 

Something like:

```
React.useEffect(() => {
    if (colorMode) {
      toast.update('someToastId', {
        title: "Test Toast",
        status: "error",
        duration: 9000000000,
        isClosable: false
      });
    }
  }, [colorMode]);
```

Though I suppose it would be nice for this to be handled behind the scenes.

**Comment by remjx** [comment]
@dominictwlee Ah, yes, that works: [Codesandbox with the fix](https://codesandbox.io/s/chakra-ui-toast-colormode-issue-with-toast-update-1iemk?file=/src/TestToast.tsx)

How did you know there was a `toast.update()` method? It is not mentioned in the [Toast docs](https://chakra-ui.com/docs/feedback/toast)

I agree that it should be handled by Chakra natively. So I think we should keep this issue open.

**Comment by gejimayu** [comment]
can i help with this ?

**Comment by remjx** [comment]
@stalebot keep open

**Comment by ivanjeremic** [comment]
I have created a custom hook to solve this problem and to reuse toasts easily.

```
import { useEffect } from 'react';
import { useColorMode, useToast } from '@chakra-ui/react';

export const useBetterToast = (settings: any) => {
  const toast = useToast();
  const { colorMode } = useColorMode();

  const { id } = settings;

  useEffect(() => {
    if (colorMode) {
      toast.update(id, settings);
    }
  }, [colorMode]);

  return () => toast(settings);
};
```

**And then use it in your Component how often you want**

```
import React from 'react';
import { useBetterToast } from './hooks/useBetterToast';

function Home() {
  // Toast 1
  const errToast = useBetterToast({
    id: 'toast1',
    title: 'API authentication failed',
    status: 'error',
    isClosable: true,
    duration: 8000,
  });

  // Toast 2
  const toastSuccess = useBetterToast({
    id: 'toast2',
    title: 'Success',
    status: 'success',
    isClosable: true,
    duration: 8000,
  });

  return (
    <div>
      <button onClick={() => errToast()}>Toast 1</button>
      <button onClick={() => toastSuccess()}>Toast 2</button>
    </div>
  );
}

```

**Comment by remjx** [comment]
Reply from Ivan Jeremic (showed up in my e-mail but not here, so copy & pasting below):

> I have created a custom hook to solve this problem and to reuse toasts easily.

```
import { useEffect } from 'react';
import { useColorMode, useToast } from '@chakra-ui/react';

export const useBetterToast = (settings: any) => {
  const toast = useToast();
  const { colorMode } = useColorMode();

  const { id } = settings;

  useEffect(() => {
    if (colorMode) {
      toast.update(id, settings);
    }
  }, [colorMode]);

  return () => toast(settings);
};
```

> And then use it in your Component how often you want

```
import React from 'react';
import { useBetterToast } from './hooks/useBetterToast';

function Home() {
  // Toast 1
  const errToast = useBetterToast({
    id: 'toast1',
    title: 'API authentication failed',
    status: 'error',
    isClosable: true,
    duration: 8000,
  });

  // Toast 2
  const toastSuccess = useBetterToast({
    id: 'toast2',
    title: 'Success',
    status: 'success',
    isClosable: true,
    duration: 8000,
  });

  return (
    <div>
      <button onClick={() => errToast()}>Toast 1</button>
      <button onClick={() => toastSuccess()}>Toast 2</button>
    </div>
  );
}
```

**Comment by ivanjeremic** [comment]
> Reply from @ivanjeremic (showed up in my e-mail but not here, so copy & pasting below):
> 
> > I have created a custom hook to solve this problem and to reuse toasts easily.
> 
> ```
> import { useEffect } from 'react';
> import { useColorMode, useToast } from '@chakra-ui/react';
> 
> export const useBetterToast = (settings: any) => {
>   const toast = useToast();
>   const { colorMode } = useColorMode();
> 
>   const { id } = settings;
> 
>   useEffect(() => {
>     if (colorMode) {
>       toast.update(id, settings);
>     }
>   }, [colorMode]);
> 
>   return () => toast(settings);
> };
> ```
> 
> > And then use it in your Component how often you want
> 
> ```
> import React from 'react';
> import { useBetterToast } from './hooks/useBetterToast';
> 
> function Home() {
>   // Toast 1
>   const errToast = useBetterToast({
>     id: 'toast1',
>     title: 'API authentication failed',
>     status: 'error',
>     isClosable: true,
>     duration: 8000,
>   });
> 
>   // Toast 2
>   const toastSuccess = useBetterToast({
>     id: 'toast2',
>     title: 'Success',
>     status: 'success',
>     isClosable: true,
>     duration: 8000,
>   });
> 
>   return (
>     <div>
>       <button onClick={() => errToast()}>Toast 1</button>
>       <button onClick={() => toastSuccess()}>Toast 2</button>
>     </div>
>   );
> }
> ```

I deleted it again because I had no time to test it to be sure everything works, but if people want they can try it out and play around with the hook.

**Comment by remjx** [comment]
Bump

**Comment by anubra266** [comment]
@remjx, you're still having this issue?

**Comment by dufia** [comment]
bump

**Comment by jls5** [comment]
I'm using a custom component for my toasts and was having the same issue with the useColorModeValue-hook. Fixed it by using the _light and _dark props of the Box-component: 
`
_light={{background: "white"}}
_dark={{background: "black"}}
`

instead of: 

`
bg={{useColorModeValue("white", black"")}}
`

**Comment by Jamess-Lucass** [comment]
+1 This would be nice for it to be handled behind the scenes for us

**Comment by segunadebayo** [comment]
This issue has been fixed in v2.  Please update to the most recent version to resolve the problem.

If after the upgrade the problem still exists in your application please open a new issue and follow the instructions in the issue template.

**Comment by anubra266** [comment]
> I also had this problem in v2.2.3, and found the cause... (User error :P) Just putting this here in case anyone has the same problem as me.
> 
> ### TLDR:
> _Make sure the ColorModeProvider is a parent of the ChakraProvider._
> 
> So my initial entry for my react app looked like this:
> 
> ```ts
> ReactDOM.createRoot(rootElement).render(
>   <StrictMode>
>     <ColorModeScript initialColorMode={theme.config.initialColorMode} />
>     <ChakraProvider theme={theme}>
>       <ColorModeProvider>
>         <Dashboard />
>       </ColorModeProvider>
>     </ChakraProvider>
>   </StrictMode>
> );
> ```
> 
> everything worked fine, except portalled content would not follow changes in color mode. they would be set permanently to the color mode active on tthe first render.
> 
> I then changed the code to look like this:
> 
> ```ts
> ReactDOM.createRoot(rootElement).render(
>   <StrictMode>
>     <ColorModeScript initialColorMode={theme.config.initialColorMode} />
>     <ColorModeProvider>
>       <ChakraProvider theme={theme}>
>         <Dashboard />
>       </ChakraProvider>
>     </ColorModeProvider>
>   </StrictMode>
> );
> ```
> 
> and everything now works correctly. :)
> 
> It seems the ChakraProvider is required to be a child of the ColorModeProvider, which i had around the wrong way initially.

@thtliife Have you tried removing `ColorModeProvider` all together. I don't think it's needed.

**Comment by thtliife** [comment]
@anubra266 you are right... turns out i dont need it at all...
I was only using it because of the readme in here:
https://github.com/chakra-ui/chakra-ui/blob/790d2417a3f5d59e2d69229a027af671c2dc0cbc/packages/color-mode/README.md

Guess ill be removing it now. Thanks mate 👏🏼
